### PR TITLE
script api documentation

### LINKED ISSUE #424: [DOC] Add missing Core REST API endpoints
Because the OpenSearch REST API documentation is no longer automated, we need to create a feature complete API reference, one the includes all endpoints, request bodies, and responses. 

For reference on what ideal REST API reference page looks like, see the [OpenSearch Create Index API](https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/create-index/).

## List of missing endpoints

To assist in achieving this goal, I've put together a list of APIs that are missing, based on cross-referencing the old [Open Distro API reference](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/). NOTE: This list is not comprehensive and might not include all missing endpoints.

- [x] [mget](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#mget)
- [x] [nodes.hot_threads](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#nodeshot_threads)
- [x] [nodes.info](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#nodesinfo)
- [x] [node.reload](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#nodesreload_secure_settings) - Need to document this (only available in OpenDistro doc)
- [x] [nodes.stats](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#nodesreload_secure_settings) https://github.com/opensearch-project/documentation-website/pull/1369
- [x] [nodes.usage](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#nodesusage) - Need to document this (only available in OpenDistro doc)
- [x] [put.script](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#nodesusage)
- [x] [rank.eval](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#rank_eval)
- [x] [snapshot.cleanup](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotcleanup_repository)
- [x] [snapshot.clone](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotclone)
- [x] [snapshot.create](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotcreate) | PR #873 
- [x] [snapshot.create_repository](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotcreate_repository)
- [x] [snapshot.delete](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotdelete)
- [x] [snapshot.delete_repository](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotdelete_repository)
- [x] [snapshot.get](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotget)
- [x] [snapshot.get_repository](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotget_repository)
- [x] [snapshot.restore](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotrestore)
- [x] [snapshot.status](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotstatus)
- [x] [snapshot.verify_repository](https://opendistro.github.io/for-elasticsearch-docs/docs/elasticsearch/rest-api-reference/#snapshotverify_repository)
- [x] `GET _scripts/<script-id>` | #1016 
- [x]  `PUT _scripts/<script-id>` | #1016 
- [x]  `PUT _scripts/<script-id>/<context>` | #1016 
- [x]  `DELETE _scripts/<script-id>` | #1016 
- [x] `GET _script_context` | #1016 
- [x] `GET _script_language` | #1016 
- [x] `POST /_scripts/painless/_execute` | #1016 
- [x] `POST /<target>/_reload_search_analyzers` | Not in code base
- [x] `GET /<target>/_reload_search_analyzers` | Not in code base
- [x] `GET /_analyze` https://github.com/opensearch-project/documentation-website/pull/1373/files
- [x] `POST /_analyze` https://github.com/opensearch-project/documentation-website/pull/1373/files
- [x] `GET /<index>/_analyze`https://github.com/opensearch-project/documentation-website/pull/1339
- [x] `POST /<index>/_analyze` https://github.com/opensearch-project/documentation-website/pull/1339
- [x] `POST /<target>/_cache/clear` https://github.com/opensearch-project/documentation-website/pull/1168
- [x] `POST /_cache/clear` https://github.com/opensearch-project/documentation-website/pull/1168
- [x] `DELETE /_dangling/<index-uuid>` https://github.com/opensearch-project/documentation-website/pull/1370
- [x] `POST /_dangling/<index-uuid>` https://github.com/opensearch-project/documentation-website/pull/1370
- [x] [ism_rollup](https://forum.opensearch.org/t/rollup-action-in-ism/5482) https://github.com/opensearch-project/documentation-website/pull/1371

After the request and responses have been documented, please check off the API from this list.

## Test API 

1. Install OpenSearch. We recommend using [Docker](https://opensearch.org/docs/latest/opensearch/install/docker/) with the a sample Docker compose file that [disables the security plugin](https://opensearch.org/docs/latest/opensearch/install/docker/#sample-docker-compose-file-for-development).
2. After installation, run OpenSearch and log into Dashboards.
3. Find the API test console. There, enter each API endpoint listed above with the correct request body. Upon successful execution, the response appears on the right column.

## Time to complete

Each API endpoint should only take about 4 - 5 hours to test, document, and address any review feedback required. With 19 found undocumented endpoints, this assumes an average time to complete of **12 - 15 days**.

### PR DESCRIPTION
**Pull request by ariamarble**
### Description
new pull request replacing #1016 

### Issues Resolved
Partially resolves #424

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ariamarble** [comment]
tech review has been completed. api endpoints and scripts verified working

File: _opensearch/rest-api/_script-apis/index.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+redirect_from:
+  - /opensearch/rest-api/script-apis/
+---
+
+# Script APIs
+
+The script APIs allow you to work with stored scripts. Stored scripts are part of the cluster state and reduce compilation time and enhance search speed. The default scripting language is Painless. 
+
+You perform the following operations on stored scripts:
+* [create]({{site.url}}{{site.baseurl}}/opensearch/rest-api/script-apis/../../../../create-stored-script/)
```
Comment (kolchfa-aws) [inline _opensearch/rest-api/_script-apis/index.md:16]: Please delete the ../../../..

Comment (ariamarble) [reply _opensearch/rest-api/_script-apis/index.md]: done

File: _opensearch/rest-api/_script-apis/create-stored-script.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+| script-id | String | Stored script or search template ID. Must be unique across the cluster. Required. |
+
+### Query parameters
+
+All parameters are optional.
+
+| Parameter | Data Type | Description | 
+:--- | :--- | :---
+| context | String | Context in which the script or search template is to run. To prevent errors, the API immediately compiles the script or template in this context. |
+| cluster_manager_timeout | Time | Amount of time to wait for a connection to the master node. Defaults to 30 seconds. |
```
Comment (cwillum) [inline _opensearch/rest-api/_script-apis/create-stored-script.md:33]: We need to decide what term to use to replace "master node" (in running text). Primary node? We use "primary" and secondary shard.

**Review by alicejw-aws — APPROVED** [review]
Looks good!

File: _opensearch/rest-api/_script-apis/create-stored-script.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+| Field | Data Type | Description | 
+:--- | :--- | :---
+| script | Object | Defines the script or search template, its parameters, and its language. See *Script object* below. |
+
+*Script object*
+
+| Field | Data Type | Description | 
+:--- | :--- | :---
+| lang | String | Scripting language. Required. |
+| source | String or Object | Required. <br />  <br /> For scripts, a string with the contents of the script. <br />  <br /> For search templates, an object that defines the search template. Supports the same parameters as the [Search API]({{site.url}}{{site.baseurl}}/opensearch/rest-api/search)'s request body. request body. Search templates also support Mustache variables. |
```
Comment (cwillum) [inline _opensearch/rest-api/_script-apis/create-stored-script.md:47]: "request body" repeated in this line?

File: _opensearch/rest-api/_script-apis/create-stored-script.md (hunk 2 above)
Comment (alicejw-aws) [reply _opensearch/rest-api/_script-apis/create-stored-script.md:33]: oh, actually the new node name is "cluster_manager" node.

Comment (ariamarble) [reply _opensearch/rest-api/_script-apis/create-stored-script.md:33]: yeah primary and secondary are best I think. I'll go ahead and update

File: _opensearch/rest-api/_script-apis/create-stored-script.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+      "source": "\n          int total = 0;\n          for (int i = 0; i < doc['\''ratings'\''].length; ++i) {\n            total += doc['\''ratings'\''][i];\n          }\n          return total;\n        "
+  }
+}'
+````
+
+See [Execute Painless stored script]({{site.url}}{{site.baseurl}}/opensearch/rest-api/_script-apis/exec-stored-script) for information about running the script.
+
+#### Sample response
+
+The `PUT _scripts/my-first-script` request returns the following fields:
```
Comment (cwillum) [inline _opensearch/rest-api/_script-apis/create-stored-script.md:99]: a single field or plural?

File: _opensearch/rest-api/_script-apis/create-stored-script.md (hunk 3 above)
Comment (ariamarble) [reply _opensearch/rest-api/_script-apis/create-stored-script.md:47]: fixed

File: _opensearch/rest-api/_script-apis/create-stored-script.md (hunk 4 above)
Comment (ariamarble) [reply _opensearch/rest-api/_script-apis/create-stored-script.md:99]: singular. corrected

File: _opensearch/rest-api/_script-apis/delete-script.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+| Parameter | Data Type | Description | 
+:--- | :--- | :---
+| script-id | String | ID of script to delete. |
+
+### Query parameters
+
+| Parameter | Data Type | Description | 
+:--- | :--- | :---
+| cluster_manager_timeout | Time | Amount of time to wait for a connection to the master node. Optional, defaults to `30s`. |
+| timeout | Time | The period of time to wait for a response. If a response is not received before the timeout value, the requ
```
Comment (cwillum) [inline _opensearch/rest-api/_script-apis/delete-script.md:26]: Is this last sentence clipped?

File: _opensearch/rest-api/_script-apis/get-stored-script.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+| Parameter | Data Type | Description | 
+:--- | :--- | :---
+| script | String | Stored script or search template name. Required.|
+
+### Query parameters
+
+| Parameter | Data Type | Description | 
+:--- | :--- | :---
+| cluster_manager_timeout | Time | Amount of time to wait for a connection to the master node. Optional, defaults to `30s`. |
```
Comment (cwillum) [inline _opensearch/rest-api/_script-apis/get-stored-script.md:23]: "master node" issue again.

File: _opensearch/rest-api/_script-apis/delete-script.md (hunk 5 above)
Comment (ariamarble) [reply _opensearch/rest-api/_script-apis/delete-script.md:26]: fixed

File: _opensearch/rest-api/_script-apis/get-stored-script.md (hunk 6 above)
Comment (ariamarble) [reply _opensearch/rest-api/_script-apis/get-stored-script.md:23]: updating these all to "cluster manager"

**Review by cwillum — COMMENTED** [review]
Looks great.

**Pull request merged**